packages/manager/apps/pci-workflow/src/api/hooks/order.ts

```diff
@@ -3,53 +3,66 @@ import {
   getCatalogQuery,
   getProductAvailabilityQuery,
 } from '@ovh-ux/manager-pci-common';
+import { useMemo } from 'react';
+import { useMe } from '@ovh-ux/manager-react-components';
 import { isSnapshotConsumption } from '@/pages/new/utils/is-snapshot-consumption';
 
-export const useProjectAddons = (
-  projectId: string,
-  ovhSubsidiary: string,
-  instanceRegion: string,
-) =>
-  useQueries({
+export const useProjectSnapshotAddons = (projectId: string) => {
+  const { me } = useMe();
+  return useQueries({
     queries: [
-      getCatalogQuery(ovhSubsidiary),
-      getProductAvailabilityQuery(projectId, ovhSubsidiary),
+      { ...getCatalogQuery(me?.ovhSubsidiary), enabled: !!me?.ovhSubsidiary },
+      {
+        ...getProductAvailabilityQuery(projectId, me?.ovhSubsidiary, {
+          addonFamily: 'snapshot',
+        }),
+        enabled: !!me?.ovhSubsidiary,
+      },
     ],
 
     combine: ([
       { data: catalog, isFetching: isCatalogFetching },
-      { data: plans, isFetching: isPlansFetching },
+      { data: snapshotAvailabilities, isFetching: isPlansFetching },
     ]) => {
-      const plan = plans?.plans?.find(
-        ({ code, regions }) =>
-          isSnapshotConsumption(code) &&
-          regions.find(({ name }) => name === instanceRegion),
-      );
-      const addons = [
-        catalog?.addons.find(({ planCode }) => planCode === plan?.code),
-      ];
-      const isFetching = isCatalogFetching || isPlansFetching;
+      const snapshotRegionPlans =
+        snapshotAvailabilities?.plans?.filter(({ code }) =>
+          isSnapshotConsumption(code),
+        ) ?? [];
 
-      return { addons, isFetching };
+      return {
+        addons: catalog?.addons
+          .filter(({ planCode }) =>
```

Comment:
nitpick: filter + map + 2 find (that does the same search), I think you could optimize it, for example using reduce

Reply:
I will check